### ISSUE TITLE
Shader Include: Can't change shader_type

### ISSUE THREAD

**Issue by HaruYou27** [godotengine/godot#98885]
### Tested versions

v4.3.stable.arch_linux
v.4.0.stable

### System information

Godot v4.3.stable unknown - Arch Linux #1 ZEN SMP PREEMPT_DYNAMIC Fri, 01 Nov 2024 03:30:35 +0000 - Wayland - GLES3 (Compatibility) - Mesa Intel(R) HD Graphics 3000 (SNB GT2) - Intel(R) Core(TM) i7-2670QM CPU @ 2.20GHz (8 Threads)

### Issue description

By default, shader include type is *spatial*, which change the type of *built-ins*
![Screenshot_20241106_133627](https://github.com/user-attachments/assets/e234427e-7bf8-481e-80e0-b6a442f8812e)
However, it is not possible to change the *shader_type* of shader include file.
![Screenshot_20241106_133604](https://github.com/user-attachments/assets/5e02f5f8-a815-430f-ae87-58ad6e38893f)
Would this be a problem if I ```#include``` in a ```canvas_item``` shader?
Also, is this mean that I can't include built-ins function like vertex, fragment, light?

### Steps to reproduce

Create a shader include file.
Try to change shader type.

### Minimal reproduction project (MRP)

[shaderincludeissue.zip](https://github.com/user-attachments/files/17642755/shaderincludeissue.zip)

**Comment by clayjohn** [comment]
If I recall correctly, the allowed syntax within a shader include file comes from the type of file in which it is included. Have you tried including the shaderinclude file from a canvas item shader?

**Comment by tetrapod00** [comment]
Yeah, it seems like the actual shader will compile fine, but the shader include will show a compile error if it is open.

`test_inc.gdshaderinc`
```
varying vec2 v;
void vertex()
{
	v = VERTEX; // VERTEX should be a vec2 in a CanvasItem shader.
}
```

`canvas_test.gdshader`:
```
shader_type canvas_item;

#include "res://test_inc.gdshaderinc"
// Includes this:
//varying vec2 v;
//void vertex()
//{
	//v = VERTEX; // VERTEX should be a vec2 in a CanvasItem shader.
//}

void fragment() {
	COLOR = vec4(v, 0.0, 1.0);
}
```

The sprite shows up fine, and the canvas item shader compiles fine with no errors.
![Godot_v4 3-stable_win64_fWisSCpdHL](https://github.com/user-attachments/assets/f2870875-1250-42c1-9e88-38a5c2be1244)
However, the shader include file has a compile error when the file is open.
![Godot_v4 3-stable_win64_B5CGt5LL2a](https://github.com/user-attachments/assets/5981770c-60b1-4fda-8fa7-22f54562bee3)


There is probably *some* documentation improvement that could be made in the [#include](https://docs.godotengine.org/en/stable/tutorials/shaders/shader_reference/shader_preprocessor.html#include) section.

There are probably also some improvements that could be made to the either the preprocessor or the way shader include files are compiled and displayed in the shader editor. 

But it seems like the basic functionality here is fine, it just has some compile errors (in the editor) that can be safely ignored.

**Comment by MGilleronFJ** [comment]
I had similar issues when using shader includes in a project where I had to implement shader variants, kept getting pointless errors which made development annoying, which led me to pass everything via functions https://github.com/Zylann/godot_atmosphere_shader/blob/330cfe9da1f579dab0677027def9ac3e2563a847/addons/zylann.atmosphere/shaders/include/planet_atmosphere_main.gdshaderinc#L5

**Comment by HaruYou27** [comment]
> If I recall correctly, the allowed syntax within a shader include file comes from the type of file in which it is included. Have you tried including the shaderinclude file from a canvas item shader?

![Screenshot_20241106_190958](https://github.com/user-attachments/assets/568806ae-1054-4ffe-be80-7f86555cc9f3)
No complie error, the shader works correctly. But the editor thinks otherwise.
